File: client-sdk/spice-java-sdk-sample/src/main/java/ai/spice/example/Cloud.java
package ai.spice.example;

import java.net.URI;

import org.apache.arrow.flight.FlightStream;
import org.apache.arrow.vector.VectorSchemaRoot;

import ai.spice.SpiceClient;

public class Cloud {
    public static void main(String[] args) {
        try (
            SpiceClient client = SpiceClient.builder()
                .withApiKey("API_KEY")
                .withHttpAddress(URI.create("https://data.spiceai.io"))
                .withFlightAddress(URI.create("grpc+tls://flight.spiceai.io:443"))
                .build()
        ) {
            FlightStream stream = client.query(
                "show tables;"
            );

            while (stream.next()) {
                try (VectorSchemaRoot batches = stream.getRoot()) {
                    System.out.println(batches.contentToTSVString());
                }
            }
        } catch (Exception e) {
            System.err.println("An unexpected error occurred: " + e.getMessage());
        }
    }
}

File: client-sdk/spice-dotnet-sdk-sample/Cloud.cs
using Spice;
using System.Threading.Tasks;

public class Cloud
{
	public static async Task Main(string[] args)
	{
		var client = new SpiceClientBuilder()
            .WithApiKey("API_KEY")
            .WithHttpAddress("https://data.spiceai.io")
            .WithFlightAddress("flight.spiceai.io:443")
            .WithTls(true)
			.Build();
		var data = await client.Query("show tables;");
	}
}
